Kernel/Devices/MBRPartitionTable.cpp

```diff
@@ -0,0 +1,74 @@
+#include <AK/ByteBuffer.h>
+#include <Kernel/Devices/MBRPartitionTable.h>
+
+#define MBR_DEBUG
+
+Retained<MBRPartitionTable> MBRPartitionTable::create(Retained<DiskDevice>&& device)
+{
+    return adopt(*new MBRPartitionTable(move(device)));
+}
+
+MBRPartitionTable::MBRPartitionTable(Retained<DiskDevice>&& device)
+    : m_device(move(device))
+{
+}
+
+MBRPartitionTable::~MBRPartitionTable()
+{
+}
+
+ByteBuffer MBRPartitionTable::read_header() const
+{
+    auto buffer = ByteBuffer::create_uninitialized(512);
```

Comment:
Since we never free this, let's just use an inline `byte[]` member in MBRPartitionTable instead. That will save us 512 bytes of kernel heap memory.